janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/SolrIndex.java

```diff
@@ -239,11 +240,24 @@ public SolrIndex(final Configuration config) throws BackendException {
         waitSearcher = config.get(WAIT_SEARCHER);
 
         if (mode==Mode.CLOUD) {
-            final String zookeeperUrl = config.get(SolrIndex.ZOOKEEPER_URL);
-            final CloudSolrClient cloudServer = new CloudSolrClient.Builder()
-                .withZkHost(zookeeperUrl)
-                .sendUpdatesOnlyToShardLeaders()
-                .build();
+            final String[] zookeeperUrl = config.get(SolrIndex.ZOOKEEPER_URL);
+            // Process possible zookeeper chroot. e.g. localhost:2181/solr
+            String chroot = null;
+            for (int i = 0; i < zookeeperUrl.length; i++) {
+                if (zookeeperUrl[i].contains("/")) {
+                    int chrootIndex = zookeeperUrl[i].indexOf("/");
+                    String hostAndPort = zookeeperUrl[i].substring(0, chrootIndex);
+                    chroot = zookeeperUrl[i].substring(chrootIndex);
```

Comment:
Is different `chroot` possible? I.e. something like this?:
```
localhost:2181/solr
localhost:2181/mysolr
localhost:2181/mysolr2
```
If yes, the you rewrite your `chroot` here each time. If you need only the last `chroot` you can do a reverse loop (i.e. `for (int i = zookeeperUrl.length-1; i >= 0 ; i--)`) and then check `if(chroot==null){chroot = zookeeperUrl[i].substring(chrootIndex)}`.

Reply:
Also, it's possible to do only one search for "/".
```
int chrootIndex = zookeeperUrl[i].indexOf("/");
if(chrootIndex!=-1){
...
}
```
But both 2 optimizations (reverse loop and one search) are not critical at all. I'm more interested if different `chroot` values possible or they all the same?

Reply:
Different chroot is not supported.
I was trying to allow the chroot to be specified in any of the strings. Bu I like your suggestion.  Will update it.